moz/sm-opt

```diff
@@ -0,0 +1,24 @@
+# Build only the JS shell
+ac_add_options --enable-application=js
+
+# Enable optimization for speed
+ac_add_options --enable-optimize
+
+# Disable debug checks to better match a release build of Firefox.
+ac_add_options --disable-debug
+
+ac_add_options --disable-ctypes
+ac_add_options --disable-jit
+ac_add_options --disable-jemalloc
+ac_add_options --enable-hardening
+ac_add_options --with-intl-api
+ac_add_options --build-backends=RecursiveMake
+#ac_add_options --with-system-icu
```

Comment:
The "problem" with using `--with-system-icu` is, that it want to use PKG_CONFIG chain - even in windows. I don't know how to prevent this under windows...

Reply:
I can't be sure in the mach build but on legacy builds, doing `choco install pkgconfiglite` causes the attempt to use pkgconf to non-fatal (but not be helpful at finding ICU), while the env variables (PATH,LIB,INCLUDE) set in shell.ps1 make icu directly discoverable by the build.

Reply:
Summary of our discussion and observations from Slack:

1. Compiling sm without `--with-system-icu`builds all icu functions into the mozjs-dll, with no external dependencies to the icu lib
2. Far fewer problems compiling the SpiderMonkey library (under windows), like no need for pkgconfiglite, setting different env vars, some in PS, some in the Mozilla build env 
3. link the CouchDB nifs against the icu lib from VCPKG (like before)
4. ATM, no problems at the moment between the external ICU library and the integrated SpiderMonkey ICU functionality